bootstrap-salt.sh

```diff
@@ -1765,12 +1765,25 @@ __function_defined() {
 }
 
 
+#---  FUNCTION  -------------------------------------------------------------------------------------------------------
+#          NAME:  __wait_for_apt
+#   DESCRIPTION:  Check if any apt process is running before calling apt-get again. This is
+#                 useful when an apt-get call is part of a boot process and will wait until
+#                 wait until the apt-get process is finished, such as on AWS AMIs.
+#----------------------------------------------------------------------------------------------------------------------
+__wait_for_apt(){
+    while pgrep -l apt; do
```

Comment:
The `pgrep` command is a part of `procps` package. It does not belong to `required` Debian archive section, so there's a possibility that it would not be installed by default on "tiny-core" systems.
More important, that this check does not cover raw `dpkg` operations, when it does not invoked by `apt-get` or similar tools.

I suggest to use some harmless APT operation to check if lock is in place, for example: `apt-get check 2>/dev/null`.
Also, having a timeout, even hard-coded, would be very convenient.

Reply:
Ah, I was worried about that. I did a little research to see if `pgrep` is included and the ones I looked at did have it. But I didn't think about the smaller options. I can update it to your suggestion.

For the time out, what would be useful here? I know we don't want to hang, but we also don't want to short-circuit the wait functionality.

Reply:
As for timeout I think 15 minutes (900 seconds) should be more than enough. Because longer time to wait would indicate that something really going too slow or bad things happen, so we could not help that in any case.

And I think wrapping this "waiter" function with INFO or DEBUG messages would be very useful as well. Or maybe I'm asking too much? 😃

Reply:
> Or maybe I'm asking too much? 

Not at all! Great idea. I'll fix that.